app/controllers/attachments_controller.rb

```diff
@@ -60,7 +61,11 @@ def show_as_html
   private
 
   def info_request
-    @info_request ||= InfoRequest.find(params[:id])
+    if public_token
```

Comment:
Could remove the duplication in assignment here, though perhaps a little less readable?

```ruby
  @info_request ||= begin
    InfoRequest.find_by!(public_token: public_token) if public_token
    InfoRequest.find(params[:id])
  end
```

Alternatively could do:

```ruby
@info_request ||=
  if public_token
    InfoRequest.find_by!(public_token: public_token)
  else
    InfoRequest.find(params[:id])
  end
```
Don't feel strongly about any of these though – just suggestions.

Reply:
We don't have any specs for accessing attachments via a `public_token` though – I think we should.